src/sketch.js

```diff
@@ -48,110 +84,201 @@ export default function sketch(p) {
                 }
             }
-    }
+        }
 
     // p.windowResized = function () {
     //     p.resizeCanvas(p.windowWidth * (8 / 12), p.windowHeight);
     // };
 
     p.mouseWheel = function (event) {
-        if (p.mouseX < 0 || p.mouseX > p.width ||
-            p.mouseY < 0 || p.mouseY > p.height) return;
+        if (p.mouseX < 0
+            || p.mouseX >= p.width
+            || p.mouseY < 0
+            || p.mouseY >= p.height
+        ) {
+            return;
+        }
 
-        const adjX = (p.mouseX - pixelOffsetX);
-        const adjY = (p.mouseY - pixelOffsetY);
+        const adjX = (p.mouseX - pixelOffsetX),
+            adjY = (p.mouseY - pixelOffsetY),
+            zoomCoefficient = 1.1
+        ;
 
         let postZoomX = 0,
-            postZoomY = 0;    
-        
+            postZoomY = 0;
+
         if (event.deltaY < 0) {
-            zoomLevel *= 1.1;
-            postZoomX = adjX * 1.1;
-            postZoomY = adjY * 1.1;
-        } else {
-            zoomLevel /= 1.1;
-            postZoomX = adjX / 1.1;
-            postZoomY = adjY / 1.1;
+            zoomLevel *= zoomCoefficient;
+            postZoomX = adjX * zoomCoefficient;
+            postZoomY = adjY * zoomCoefficient;
         }
-        
+        else {
+            zoomLevel /= zoomCoefficient;
+            postZoomX = adjX / zoomCoefficient;
+            postZoomY = adjY / zoomCoefficient;
+        }
+
         pixelOffsetX += adjX - postZoomX;
         pixelOffsetY += adjY - postZoomY;
     };
 
-    p.mouseDragged = function (event) {
-        if (p.mouseX < 0 || p.mouseX > p.width ||
-            p.mouseY < 0 || p.mouseY > p.height) return;
+    // Используем свои переменные нарочно 
+    // (я не использую p.pmouseX и p.pmouseY т.к. они обновляются всегда, а не когда мне нужно)
+    let dragPrevX = -1, dragPrevY = -1; 
+    function analyzeDrag() {
+        if (dragPrevX === p.mouseX && dragPrevY === p.mouseY) {
+            return;
+        }
 
-        if (dragMode === "move") {
+        if (p.mouseX < 0
+            || p.mouseX >= p.width
+            || p.mouseY < 0
+            || p.mouseY >= p.height
+        ) {
+            return;
+        }
 
-            pixelOffsetX += p.mouseX - p.pmouseX;
-            pixelOffsetY += p.mouseY - p.pmouseY;
-        } else {
+        const adjTileSize = (tileSize * zoomLevel);
+        const adjX = (p.mouseX - pixelOffsetX);
+        const adjY = (p.mouseY - pixelOffsetY);
+
+        const currRow = p.floor(adjX / adjTileSize);
+        const currCol = p.floor(adjY / adjTileSize);
 
+        try {
+            if (dragMode === "move") {
+                p.cursor('grabbing');
+                movingCanvas = true;
+                pixelOffsetX += p.mouseX - p.pmouseX;
+                pixelOffsetY += p.mouseY - p.pmouseY;
+            } else if (dragMode === "draw") {
+                if (activeImageLabel) mapArray[currCol][currRow] = {img: activeImageLabel};
```

Comment:
```suggestion
                if (activeImageLabel) {
                    mapArray[currCol][currRow] = { img: activeImageLabel };
                }
```
See https://github.com/airbnb/javascript#whitespace--in-braces